stock_voucher/models/stock_picking.py

```diff
@@ -66,66 +65,60 @@ def get_estimated_number_of_pages(self):
         if lines_per_voucher == 0:
             estimated_number_of_pages = 1
         else:
-            operations = len(self.pack_operation_ids)
+            operations = len(self.move_lines)
             estimated_number_of_pages = int(ceil(
                 float(operations) / float(lines_per_voucher)
             ))
         return estimated_number_of_pages
 
-    @api.one
     @api.constrains('picking_type_id')
     @api.onchange('picking_type_id')
     def _get_book(self):
         self.book_id = self.picking_type_id.book_id
 
-    @api.multi
-    @api.onchange(
-        'pack_operation_ids',
-        'pack_operation_product_ids',
-        'pack_operation_pack_ids',
-    )
+    @api.constrains('move_lines')
+    @api.onchange('move_lines')
     def product_onchange(self):
         # self.declared_value = 0
+        SaleOrderLine = self.env['sale.order.line']
         for rec in self:
             if not rec.automatic_declare_value:
                 return False
```

Comment:
`return False` has not sense inside the for because will return the first time and will not evaluate the rest of the records.

Replace

```python
         for rec in self:
             if not rec.automatic_declare_value:
                 return False
```
With 

```python
         for rec in self.filtered(lambda x: not x.automatic_declare_value):
```